### PR TITLE
Nessie: Add Catalog tests for NessieCatalog

### PR DESCRIPTION
**Pull request by nastra**
currently depends on #4385 and #4333

### CODE REVIEW

**Review by snazy — COMMENTED** [review]
The last commit LGTM, but you will to rebase, so saving my "Approve" for that time ;)

File: core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java [hunk 1]
```diff
@@ -82,9 +81,8 @@
   private static JdbcCatalog catalog;
   private static String warehouseLocation;
 
-  @Rule
-  public TemporaryFolder temp = new TemporaryFolder();
-  File tableDir = null;
+  @TempDir
+  java.nio.file.Path tableDir;
```
Comment (snazy) [inline core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java:85]: Import?

**Comment by rdblue** [comment]
@nastra can you rebase this so we can make sure it still works on the current master?

File: core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java (hunk 1 above)
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java:85]: required as otherwise it would clash with the org.apache.hadoop.fs.Path import

**Comment by nastra** [comment]
@rdblue rebased the PR. Will rebase again once https://github.com/apache/iceberg/pull/4385 and https://github.com/apache/iceberg/pull/4333 are in

**Review by rymurr — APPROVED** [review]

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  }
+
+  @Override
+  protected boolean supportsNamespaceProperties() {
+    return false;
+  }
+
+  @Override
+  protected boolean supportsServerSideRetry() {
+    // TODO: we do support retries, but a bunch of tests are currently failing
```
Comment (rdblue) [inline nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java:144]: I think you'd have to update to the diff-based commits to make this work. If you end up using the REST catalog API that would be a nice feature to pull in at the same time.

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    return false;
+  }
+
+  @Test
+  @Override
+  @Disabled(
+      "Nessie does not differentiate between table creates & updates, thus a concurrent transaction does not fail")
```
Comment (rdblue) [inline nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java:151]: Not even in the same branch?

**Comment by rdblue** [comment]
Thanks for reviewing, @rymurr! Sorry I didn't get to this.

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java (hunk 3 above)
Comment (nastra) [reply nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java:151]: currently no, not even on the same branch. However, it's likely that we add this in an upcoming release of Nessie

Comment (rdblue) [reply nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java:151]: I'm glad that we're testing to find the differences between catalogs! Can we document this limitation somewhere? If we're disabling a test for transactional behavior then it seems like something we should note for users.

Comment (nastra) [reply nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java:151]: done in https://github.com/apache/iceberg/pull/4501

**Pull request merged**